### PR TITLE
Add more documentation around e2e and benchmarks

### PR DESCRIPTION
**Pull request by findkim**

### CODE REVIEW

File: e2e/command_test.go [hunk 1]
```diff
@@ -16,8 +17,10 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestE2E_MetaCOmmandErrors(t *testing.T) {
-	// test cases that cross subcommands that coded in the command meta object
+// TestE2E_MetaCommandErrors tests cases that cross subcommands coded in
+// the command meta object. This starts up a local Consul server and runs
+// CTS in dev mode.
+func TestE2E_MetaCommandErrors(t *testing.T) {
```
Comment (lornasong) [inline e2e/command_test.go:23]: Thanks for fixing this COmmand typo!

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // +build e2e
 
+//
```
Comment (lornasong) [inline e2e/e2e_test.go:3]: Unsure if you were maybe planning to write a comment here?

File: e2e/e2e_test.go [hunk 3]
```diff
@@ -86,9 +96,10 @@ func TestE2ERestartSync(t *testing.T) {
 	delete()
 }
 
+// TestE2ERestartConsul tests CTS is able to reconnect to Consul after the
+// Consul agent had restartsand, and CTS resumes monitoring changes to the
+// Consul catalog.
```
Comment (lornasong) [inline e2e/e2e_test.go:101]: "restartsand" => "restarted" maybe?

**Review by lornasong — APPROVED** [review]
Thanks for writing up the readme and adding comments to all these tests ❤️

File: e2e/e2e_test.go (hunk 2 above)
Comment (findkim) [reply e2e/e2e_test.go:3]: I wanted to write a high-level summary of what the file tests, but didn't come up with one and left this on accident. Thanks will remove

**Pull request merged**